### PR TITLE
include generated files

### PR DESCRIPTION
**Pull request by FreddieGilbraith**
Some projects require the generated files to be present in a tree-sitter repository to use them. Noteably the [helix editor](https://github.com/helix-editor/helix/blob/1e7c01d75be649c307636c8ad98e5c7de8a02925/languages.toml#L1708) is currently refferencing [a fork](https://github.com/jaredramirez/tree-sitter-rescript) of this grammar and therefore can not be updated to the newest version.

Please can we include the built files in the repo.

I've not included a Github Actions check to ensure the generated files are up-to-date, but am happy to add one if you'd like.

Thanks :)

### CODE REVIEW

**Comment by ribru17** [comment]
[`nvim-treesitter`](https://github.com/nvim-treesitter/nvim-treesitter) would also like this to get merged! :)

File: binding.gyp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+      "dependencies": [
+        "<!(node -p \"require('node-addon-api').targets\"):node_addon_api_except",
+      ],
+      "include_dirs": [
+        "src",
+      ],
+      "sources": [
+        "bindings/node/binding.cc",
+        "src/parser.c",
+        # NOTE: if your language has an external scanner, add it here.
```
Comment (WhyThat) [inline binding.gyp:14]: We need to add the parser here no ?

**Review by WhyThat — COMMENTED** [review]
Hello thanks for this PR should be very usefull :)

**Comment by FreddieGilbraith** [comment]
I didn't want to add [94bdab2](https://github.com/rescript-lang/tree-sitter-rescript/pull/245/commits/94bdab24303c2c84218f350d55eed9b76a759f6a), but without it [corepack](https://github.com/nodejs/corepack) was installing yarn v4

**Comment by Emilios1995** [comment]
Our motivation for keeping generated files out of the repo is aligned with [this](https://github.com/alex-pinkus/tree-sitter-swift?tab=readme-ov-file#where-is-your-parserc) explanation from the swift parser's maintainer. 

Neovim doesn't need generated files at all; it can generate them itself. Please look at the installation instructions on the readme.

Regarding helix, I personally much prefer what the Swift parser did, which is to have a branch that is automatically [updated](https://github.com/alex-pinkus/tree-sitter-swift/blob/03af4d057afc56edf6a703e6606b86f782353f22/scripts/write-generated-grammar.sh#L4) with the generated files whenever the main branch changes. (via a GH action.)

Is anyone interested in implementing copying that here?

**Comment by clason** [comment]
> Neovim doesn't need generated files at all; it can generate them itself.

Well, that's not strictly true. The nvim-treesitter plugin can call `tree-sitter` to generate the parser before compiling it, but only if the user has the tree-sitter CLI installed (which is not a hard requirement since the vast majority of grammars do commit the parser).

And it is true that upstream recommends not committing the parser **iff** you make versioned releases which _do_ contain all generated files. I (and upstream) _strongly_ recommend against having a separate branch for that; that just creates confusion and additional work. (The swift parser is not a good model to follow in general; they have very specific needs and workflows.)

Even if you don't keep the parser in the repo, though, you should update *and fix* the files you do track (bindings, headers, an in particular the scanner includes).

**Comment by Emilios1995** [comment]
Ok so we'll probably merge this, but we _would_ like a GH action to ensure generated files are up to date. @aspeddro Pointed me to this [example](https://github.com/rescript-association/rescript-core/blob/main/.github/workflows/check-build.yml) on the `rescript-core` repo. @FreddieGilbraith I'd like to take you up on your offer of adding the action, if you're still interested. 

Just one question: Is it necessary to include the bindings? AFACT from looking at @ribru17's PR in nvim-treesitter, neovim doesn't need them. Is that the case for Helix too? 

If not, we can generate the files with ` --no-bindings` and end up with smaller diffs.

**Comment by clason** [comment]
nvim-treesitter needs no bindings (Helix might need the Rust ones, I don't know), but these of course are not the only consumers. I can't speak for any of them, though.

**Comment by FreddieGilbraith** [comment]
I've added a CI check, and a git hook script to make sure generation is as part of commits.

Looks like no one needs the bindings, helix seems to work without them :)

**Comment by aspeddro** [comment]
Thank you @FreddieGilbraith!!!

**Pull request merged**